airbyte-integrations/connectors/source-hubspot/source_hubspot/api.py

```diff
@@ -589,31 +593,39 @@ def read(self, getter: Callable, params: Mapping[str, Any] = None) -> Iterator:
         payload = (
             {
                 "filters": [{"value": int(self._state.timestamp() * 1000), "propertyName": self.last_modified_field, "operator": "GTE"}],
+                "sorts": [{"propertyName": self.last_modified_field, "direction": "ASCENDING"}],
                 "properties": properties_list,
-                "limit": 100,
+                "limit": self.limit,
             }
             if self.state
             else {}
         )
 
-        while True:
-            stream_records = {}
-            if self.state:
+        after = 0
+        if self.state:
+            while after != None:
                 response = getter(data=payload)
                 for record in self._transform(self.parse_response(response)):
-                    stream_records[record["id"]] = record
-            else:
+                    yield record
+                    latest_cursor = self._field_to_datetime(record[self.updated_at_field])
+
+                after = response.get("paging", {}).get("next", {}).get("after")
+                if after and after + self.limit > self.max_search_resuls:
```

Comment:
If `after` is none this won't throw a runtime exception when trying to sum `after + self.limit`?

Reply:
we check if after exists so it should all be good:
```
>>> after = None
>>> if after and after + 1 > 0:
...     print("one")
...
>>>
```

will add some unit tests